lib/tasks/puppet.rake

```diff
@@ -57,4 +57,32 @@ namespace :puppet do
       end
     end
   end
+
+  namespace :migrate do
+    desc "
+    Migrate Puppet provisioning parameters from ENC to Host parameters.
+    This task is relevant if you want to continue using Puppet provisioning,
+    but do not want to install the Puppet plugin and get the full ENC funcionality."
+
+    task :to_parameters => :environment do
+      class FakeEnv < ::ApplicationRecord
+        self.table_name = 'environments'
+      end
+
+      User.current = User.anonymous_console_admin
+      envs = Hash[FakeEnv.all.pluck(:id, :name)]
+
+      Hostgroup.where.not(environment_id: nil).find_each(batch_size: 100) do |hg|
```

Comment:
What about HGs that don't set environment but do set puppet server or ca?

Reply:
inheriting environment does set it on the Host on `save`, doesn't it?